app/services/bulk_url_uploader.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+class BulkUrlUploader
+  MAXIMUM_FILE_SIZE = 10.megabytes
+  VALID_CONTENT_TYPES = %w[text/plain txt].freeze
+
+  attr_reader :results
+
+  class Error < StandardError
+  end
+
+  class Results
+    attr_accessor :domains, :ok_count, :error_count, :name
+
+    def initialize(name)
+      @name = name
+      @ok_count = 0
+      @error_count = 0
+      @domains = Set.new
+      @errors = Hash.new { |hash, key| hash[key] = [] }
+    end
+
+    def add_ok(url)
+      @ok_count += 1
+      @domains << url.searchgov_domain
+    end
+
+    def add_error(error_message, url)
+      @error_count += 1
+      @errors[error_message] << url
+    end
+
+    def total_count
+      ok_count + error_count
+    end
+
+    def error_messages
+      @errors.keys
+    end
+
+    def urls_with(error_message)
+      @errors[error_message]
+    end
+  end
+
+  def initialize(name, url_file)
+    @url_file = url_file
+    @results = Results.new(name)
+  end
+
+  def upload_and_index
+    upload_urls
+    index_domains
+  end
```

Comment:
All the methods other than `upload_and_index` can be private

Reply:
We can discuss this, but I seriously dislike private methods.

Reply:
Made private. Under protest.

Reply:
bump ^